.github/workflows/ci.yml

```diff
@@ -0,0 +1,52 @@
+# For more information see: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow
+
+name: CI
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  lint-and-types:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [23.x]
+    steps:
+    - uses: actions/checkout@v4
```

Comment:
nit: pin these using a sha throughout

see https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash

dependabot MIGHT fix this automatically too.

Reply:
Ah, sure!

Reply:
I don’t really think it’s important to do that on GitHub-owned actions tho.